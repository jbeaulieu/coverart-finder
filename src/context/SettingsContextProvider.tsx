import { useMemo, useState } from 'react';
import SettingsContext from './SettingsContext';
import type { ReactNode } from 'react';
import { Provider } from '../@types/enums';
import type { AppSettings } from '../@types/settings';

interface Props { children: ReactNode }

const SettingsContextProvider = (props: Props) => {
  const { children } = props;
  const [settings, setSettings] = useState<AppSettings>({ provider: Provider.iTunes });
  const value = useMemo(() => ({ settings, setSettings }), [settings]);

  return <SettingsContext value={value}>{children}</SettingsContext>;
};

export default SettingsContextProvider;
