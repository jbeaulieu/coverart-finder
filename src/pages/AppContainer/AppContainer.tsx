import SettingsIcon from '@mui/icons-material/Settings';
import App from "../../App";
import IconButton from '@mui/material/IconButton';
import Chip from '@mui/material/Chip';
import useSettingsContext from '../../context/useSettingsContext';
import { Provider, type TProvider } from '../../@types/enums';
import Avatar from '@mui/material/Avatar';
import DeezerLogo from '../../assets/deezer.png'
import iTunesLogo from '../../assets/iTunes.png';

const AppContainer = () => {
  const { settings, setSettings } = useSettingsContext();

  const setProvider = (provider: TProvider) => {
    setSettings({ ...settings, provider });
  };

  return (
    <>
      <div style={{display: 'flex', justifyContent: 'space-between'}}>
        <span>
          <Chip
            label="iTunes"
            avatar={<Avatar alt="iTunes" src={iTunesLogo} />}
            color={settings.provider == Provider.iTunes ? "primary" : "default"}
            onClick={() => setProvider(Provider.iTunes)}
            style={{marginRight: 5}}
          />
          <Chip
            label="Deezer"
            avatar={<Avatar alt="iTunes" src={DeezerLogo} />}
            color={settings.provider == Provider.Deezer ? "primary" : "default"}
            onClick={() => setProvider(Provider.Deezer)}
          />
        </span>
        <IconButton aria-label="settings">
          <SettingsIcon />
        </IconButton>
      </div>
      <App />
    </>
  );
};

export default AppContainer;
