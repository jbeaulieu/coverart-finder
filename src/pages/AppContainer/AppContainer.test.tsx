import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import SettingsContextProvider from '../../context/SettingsContextProvider';
import AppContainer from './AppContainer';

const renderAppContainer = () => {
  return render(
    <SettingsContextProvider>
      <AppContainer />
    </SettingsContextProvider>
  );
};

describe('Component AppContainer', () => {
  it('renders without crashing', () => {
    const { asFragment } = renderAppContainer();

    expect(asFragment()).toMatchSnapshot();
  });
});
