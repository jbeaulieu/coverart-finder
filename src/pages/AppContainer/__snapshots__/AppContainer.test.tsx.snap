// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component AppContainer > renders without crashing 1`] = `
<DocumentFragment>
  <div
    style="display: flex; justify-content: space-between;"
  >
    <span>
      <div
        class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-1pymu8d-MuiButtonBase-root-MuiChip-root"
        role="button"
        style="margin-right: 5px;"
        tabindex="0"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarMedium MuiChip-avatarColorPrimary css-fmeqll-MuiAvatar-root"
        >
          <img
            alt="iTunes"
            class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
            src="/src/assets/iTunes.png"
          />
        </div>
        <span
          class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
        >
          iTunes
        </span>
      </div>
      <div
        class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-lbitzw-MuiButtonBase-root-MuiChip-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarMedium MuiChip-avatarColorDefault css-fmeqll-MuiAvatar-root"
        >
          <img
            alt="iTunes"
            class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
            src="/src/assets/deezer.png"
          />
        </div>
        <span
          class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
        >
          Deezer
        </span>
      </div>
    </span>
    <button
      aria-label="settings"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="SettingsIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6"
        />
      </svg>
    </button>
  </div>
  <img
    alt="App logo"
    class="logo"
    src="/src/assets/icon.png"
  />
  <p
    class="MuiTypography-root MuiTypography-body1 css-5c82k1-MuiTypography-root"
  >
    Coverart Finder
  </p>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-152icmn-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      id="finder-container"
      style="display: flex; flex-direction: column; align-items: start;"
    >
      <form
        autocomplete="off"
        class="MuiBox-root css-7wz4ac"
        novalidate=""
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="search-container"
            id="search-container-label"
          >
            Search by Artist / Album
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-1n04w30-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-1dune0f-MuiInputBase-input-MuiOutlinedInput-input"
              id="search-container"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-elo8k2-MuiInputAdornment-root"
            >
              <button
                aria-label="search"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="submit"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                  />
                </svg>
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1n64csd-MuiNotchedOutlined-root"
              >
                <span>
                  Search by Artist / Album
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </form>
      <div
        class="MuiBox-root css-15v79bh"
      >
        <div
          style="position: relative; height: 500px; width: 500px; overflow: auto; -webkit-overflow-scrolling: touch; will-change: transform; direction: ltr;"
        >
          <div
            style="height: 216px; width: 100%;"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-19f6boz-MuiListItem-root"
              style="position: absolute; left: 0px; top: 0px; height: 72px; width: 100%;"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters Mui-selected MuiListItemButton-root MuiListItemButton-gutters Mui-selected css-4bx69l-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                  >
                    Watch the Fire
                  </span>
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
                  >
                    Boys Like Girls
                  </p>
                </div>
                <div
                  class="MuiListItemAvatar-root css-1ulznza-MuiListItemAvatar-root"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circular css-fmeqll-MuiAvatar-root"
                  >
                    <img
                      alt="Watch the Fire"
                      class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
                      src="https://is1-ssl.mzstatic.com/image/thumb/Music221/v4/c0/4b/01/c04b0132-0f23-2a9a-88f4-c22d996470e9/067003846363.png/60x60bb.jpg"
                    />
                  </div>
                </div>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters css-19f6boz-MuiListItem-root"
              style="position: absolute; left: 0px; top: 72px; height: 72px; width: 100%;"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-4bx69l-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                  >
                    Boys Like Girls
                  </span>
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
                  >
                    Boys Like Girls
                  </p>
                </div>
                <div
                  class="MuiListItemAvatar-root css-1ulznza-MuiListItemAvatar-root"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circular css-fmeqll-MuiAvatar-root"
                  >
                    <img
                      alt="Boys Like Girls"
                      class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
                      src="https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/04/5a/e4/045ae4df-461d-db63-9b6f-cb48507c682d/mzi.ddfuhxng.jpg/60x60bb.jpg"
                    />
                  </div>
                </div>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters css-19f6boz-MuiListItem-root"
              style="position: absolute; left: 0px; top: 144px; height: 72px; width: 100%;"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-4bx69l-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                  >
                    Crazy World
                  </span>
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
                  >
                    Boys Like Girls
                  </p>
                </div>
                <div
                  class="MuiListItemAvatar-root css-1ulznza-MuiListItemAvatar-root"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circular css-fmeqll-MuiAvatar-root"
                  >
                    <img
                      alt="Crazy World"
                      class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
                      src="https://is1-ssl.mzstatic.com/image/thumb/Music124/v4/d9/bd/65/d9bd655f-7de6-30a1-c1dc-9db071e9879d/886443665224.jpg/60x60bb.jpg"
                    />
                  </div>
                </div>
              </div>
            </li>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-qxh7fn"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-vjrxpc-MuiTypography-root"
          id="input-slider"
        >
          Download Size (px)
        </p>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-1bcqwu3-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-grow css-1ynczkt-MuiGrid-root"
          >
            <span
              class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-13vpke-MuiSlider-root"
            >
              <span
                class="MuiSlider-rail css-r64h58-MuiSlider-rail"
              />
              <span
                class="MuiSlider-track css-xvk2i-MuiSlider-track"
                style="left: 0%; width: 65.51724137931035%;"
              />
              <span
                class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-1xcmt9q-MuiSlider-thumb"
                data-index="0"
                style="left: 65.51724137931035%;"
              >
                <input
                  aria-labelledby="input-slider"
                  aria-orientation="horizontal"
                  aria-valuemax="1500"
                  aria-valuemin="50"
                  aria-valuenow="1000"
                  data-index="0"
                  max="1500"
                  min="50"
                  step="50"
                  style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                  type="range"
                  value="1000"
                />
              </span>
            </span>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1lair6c-MuiTypography-root"
            style="margin-bottom: 4px;"
          >
            1000
          </p>
        </div>
      </div>
    </div>
    <div
      class="preview-container"
    >
      <img
        alt="cover art preview"
        height="600"
        src="https://is1-ssl.mzstatic.com/image/thumb/Music221/v4/c0/4b/01/c04b0132-0f23-2a9a-88f4-c22d996470e9/067003846363.png/600x600bb.jpg"
        style="object-fit: cover;"
        width="600"
      />
      <div
        style="display: flex; gap: 20px; place-items: center;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Download
          <span
            class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-1gulhci-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="DownloadIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 20h14v-2H5zM19 9h-4V3H9v6H5l7 7z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
